auth-tokens/src/main/java/com/palantir/tokens/auth/BearerToken.java

```diff
@@ -80,18 +79,27 @@ public static BearerToken valueOf(String token) {
 
     // Optimized implementation of the regular expression VALIDATION_PATTERN_STRING
     private static boolean isValidBearerToken(String token) {
-        boolean isAtEnd = false;
         int length = token.length();
-        for (int i = 0; i < length; i++) {
-            char character = token.charAt(i);
-            if (isAtEnd && character != '=') {
-                return false;
-            } else if (!allowedCharacters.get(character)) {
-                return false;
+        int cursor = 0;
+
+        for (; cursor < length; cursor++) {
+            if (!allowedCharacters.get(token.charAt(cursor))) {
+                break;
```

Comment:
This feels a bit subtle, that we re-use the cursor to validate a run of `===` at the end...

Reply:
suggestions? I found this far more readable than the previous flag that determined whether we were validating the run of `==` at the end or the initial run of valid characters

Reply:
cool. I personally thought this read nicely so if you have an alternative that you think is easier to read and performant (which was the whole reason for this change) then happy to follow up